compiler/luci/pass/src/ResolveCustomOpMatMulPass.cpp

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/ResolveCustomOpMatMulPass.h"
+
+#include "flatbuffers/flexbuffers.h"
+#include <loco/IR/DataTypeTraits.h>
+
+#include <luci/IR/CircleNodes.h>
+
+#include <loco.h>
+#include <oops/InternalExn.h>
+#include <loco/Service/ShapeInference.h>
+#include <loco/Service/TypeInference.h>
+
+namespace
+{
+
+template <typename T>
+luci::CircleConst *create_const_node(loco::Graph *g, const loco::DataType dtype,
+                                     const std::vector<uint32_t> &shape,
+                                     const std::vector<T> &values)
+{
+  auto node = g->nodes()->create<luci::CircleConst>();
+  node->dtype(dtype);
+  node->rank(shape.size());
+  uint32_t size = (shape.size() == 0) ? 0 : 1;
```

Comment:
Is it possible `{a_shape.rank()}` becomes 0?

Reply:
I don't think so, this condition prevents this case: 
https://github.com/Samsung/ONE/blob/2e085c6dfd76138c4623a618db7ce3039e1efea0/compiler/luci/pass/src/ResolveCustomOpMatMulPass.cpp#L91-L92

Reply:
Lines you said appears before `create_const_node`. So I think shape always has rank more than 2.

Reply:
Ah, got it, you are right, this function is never called with an empty shape. Fixed this. Thanks!